### PR TITLE
fix(logging): improve the log output EE-2825

### PR DESCRIPTION
**Pull request by hookenz**
Closes [EE-2825](https://portainer.atlassian.net/browse/EE-2825)

### CODE REVIEW

File: compose/internal/composebinary/composebinary.go [hunk 1]
```diff
@@ -35,7 +35,12 @@ func NewComposeWrapper(binaryPath, configPath string) (libstack.Deployer, error)
 func (wrapper *ComposeWrapper) Deploy(ctx context.Context, workingDir, host, projectName string, filePaths []string, envFilePath string, forceRereate bool) error {
 	output, err := wrapper.Command(newUpCommand(filePaths, forceRereate), workingDir, host, projectName, envFilePath)
 	if len(output) != 0 {
-		log.Printf("[libstack,composebinary] [message: finish deploying] [output: %s] [err: %s]", output, err)
+		if err != nil {
+			log.Printf("[libstack,composebinary] [message: deploy complete] [output: %s] [err: %s]", output, err)
```
Comment (deviantony) [inline compose/internal/composebinary/composebinary.go:39]: any reason why you don't want the error to bubble up here? that condition looks weird:

If there is an error, log deploy complete?

I think it is not needed. If you need to handle the error, you should do it by catching the error when calling Deploy.

My comment applies to all the other occurence of this code change in this PR.

Comment (hookenz) [reply compose/internal/composebinary/composebinary.go:39]: oops, no that should be return err not return nil.  my bad.

File: compose/internal/composebinary/composebinary.go [hunk 2]
```diff
@@ -35,7 +35,12 @@ func NewComposeWrapper(binaryPath, configPath string) (libstack.Deployer, error)
 func (wrapper *ComposeWrapper) Deploy(ctx context.Context, workingDir, host, projectName string, filePaths []string, envFilePath string, forceRereate bool) error {
 	output, err := wrapper.Command(newUpCommand(filePaths, forceRereate), workingDir, host, projectName, envFilePath)
 	if len(output) != 0 {
-		log.Printf("[libstack,composebinary] [message: finish deploying] [output: %s] [err: %s]", output, err)
+		if err != nil {
+			return err
+		}
+
+		log.Printf("[INFO] [libstack,composebinary] [message: Stack deployment successful]")
```
Comment (deviantony) [inline compose/internal/composebinary/composebinary.go:42]:
```suggestion
		log.Printf("[INFO] [compose,internal,composebinary] [message: Stack deployment successful]")
```

File: compose/internal/composebinary/composebinary.go [hunk 3]
```diff
@@ -35,7 +35,12 @@ func NewComposeWrapper(binaryPath, configPath string) (libstack.Deployer, error)
...
+		log.Printf("[DEBUG] [libstack,composebinary] [output: %s]", output)
```
Comment (deviantony) [inline compose/internal/composebinary/composebinary.go:43]:
```suggestion
		log.Printf("[DEBUG] [compose,internal,composebinary] [output: %s]", output)
```

File: compose/internal/composeplugin/composeplugin.go [hunk 4]
```diff
@@ -65,7 +65,12 @@ func NewPluginWrapper(binaryPath, configPath string) (libstack.Deployer, error)
 func (wrapper *PluginWrapper) Deploy(ctx context.Context, workingDir, host, projectName string, filePaths []string, envFilePath string, forceRereate bool) error {
 	output, err := wrapper.command(newUpCommand(filePaths, forceRereate), workingDir, host, projectName, envFilePath)
 	if len(output) != 0 {
-		log.Printf("[libstack,composebinary] [message: finish deploying] [output: %s] [err: %s]", output, err)
+		if err != nil {
+			return err
+		}
+
+		log.Printf("[INFO] [libstack,composeplugin] [message: Stack deployment successful]")
```
Comment (deviantony) [inline compose/internal/composeplugin/composeplugin.go:72]:
```suggestion
		log.Printf("[INFO] [compose,internal,composeplugin] [message: Stack deployment successful]")
```

File: compose/internal/composeplugin/composeplugin.go [hunk 5]
```diff
@@ -65,7 +65,12 @@ func NewPluginWrapper(binaryPath, configPath string) (libstack.Deployer, error)
...
+		log.Printf("[DEBUG] [libstack,composeplugin] [output: %s]", output)
```
Comment (deviantony) [inline compose/internal/composeplugin/composeplugin.go:73]:
```suggestion
		log.Printf("[DEBUG] [compose,internal,composeplugin] [output: %s]", output)
```

**Review by deviantony — CHANGES_REQUESTED** [review]
Minor change to apply to all locations to both files and all occurrences and then LGTM

**Review by deviantony — APPROVED** [review]
LGTM

**Pull request merged**